src/Sylius/Behat/Context/Api/Admin/ManagingOrdersContext.php

```diff
@@ -0,0 +1,206 @@
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Client\ApiClientInterface;
+use Sylius\Behat\Client\ResponseCheckerInterface;
+use Sylius\Behat\Service\SharedStorageInterface;
+use Sylius\Component\Core\Model\CustomerInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Order\OrderTransitions;
+use Webmozart\Assert\Assert;
+
+final class ManagingOrdersContext implements Context
+{
+    /** @var ApiClientInterface */
+    private $client;
+
+    /** @var ResponseCheckerInterface */
+    private $responseChecker;
+
+    /** @var IriConverterInterface */
+    private $iriConverter;
+
+    /** @var SharedStorageInterface */
+    private $sharedStorage;
+
+    public function __construct(
+        ApiClientInterface $client,
+        ResponseCheckerInterface $responseChecker,
+        IriConverterInterface $iriConverter,
+        SharedStorageInterface $sharedStorage
+    ) {
+        $this->client = $client;
+        $this->responseChecker = $responseChecker;
+        $this->iriConverter = $iriConverter;
+        $this->sharedStorage = $sharedStorage;
+    }
+
+    /**
+     * @When I browse orders
+     */
+    public function iBrowseOrders(): void
+    {
+        $this->client->index();
+    }
+
+    /**
+     * @When I view the summary of the order :order
+     */
+    public function iSeeTheOrder(OrderInterface $order): void
+    {
+        $this->client->show($order->getNumber());
+    }
+
+    /**
+     * @When /^I cancel (this order)$/
+     */
+    public function iCancelThisOrder(OrderInterface $order): void
+    {
+        $this->client->applyTransition(
+            $this->responseChecker->getValue($this->client->show($order->getNumber()), 'number'),
+            OrderTransitions::TRANSITION_CANCEL
+        );
+    }
+
+    /**
+     * @Then I should see a single order from customer :customer
+     */
+    public function iShouldSeeASingleOrderFromCustomer(CustomerInterface $customer): void
+    {
+        Assert::true($this->responseChecker->hasItemWithValue(
+            $this->client->getLastResponse(),
+            'customer',
+            $this->iriConverter->getIriFromItem($customer)),
+            sprintf('There is no order for customer %s', $customer->getEmail())
+        );
+    }
+
+    /**
+     * @Then I should be notified that it has been successfully updated
+     */
+    public function iShouldBeNotifiedAboutItHasBeenSuccessfullyCanceled(): void
+    {
+        Assert::true(
+            $this->responseChecker->isUpdateSuccessful($this->client->getLastResponse()),
+            'Resource could not be completed'
+        );
+    }
+
+    /**
+     * @Then this order should have state :state
+     * @Then its state should be :state
+     */
+    public function itsStateShouldBe(string $state): void
+    {
+        Assert::true(
+            $this->responseChecker->hasValue(
+                $this->client->show($this->sharedStorage->get('order')->getNumber()),
```

Comment:
```suggestion
                $this->client->getLastResponse(),
```
And same usage for everything that refers to the current context?